ClientTests/URLExtensionsTests.swift

```diff
@@ -0,0 +1,28 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import Network
+
+final class URLExtensionsTests: XCTestCase {
+    func testIsIPv4() throws {
```

Comment:
Thank you for writing tests 💯 For completeness also add a few cases that have IPv4 and IPv6 in a full URL. And then also some variations that include a custom port number. That will verify the two checks work correctly with a wider range of input.